### PR TITLE
docs(recipes): discuss parsing nested multipart forms

### PR DESCRIPTION
**Pull request by vytas7**
When it comes to nested multipart forms, most other Python frameworks just slam a fat `wontfix`.

I don't think it should be something we would actively support, but our parser architecture actually happens to support some limited cases out of the box. So I went ahead and added a recipe explaining how one can enable the functionality.

### CODE REVIEW

**Comment by vytas7** [comment]
This PR also adds one small functional improvement: the deprecated `Content-Transfer-Encoding` header is accepted if its value is `binary`.

File: docs/user/recipes/multipart-mixed.rst [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+    parser = falcon.media.MultipartFormHandler()
+    parser.parse_options.media_handlers['multipart/mixed'] = (
+        falcon.media.MultipartFormHandler())
+
+.. note::
+    Here we create a new parser (with default options) for nested parts,
+    effectively disallowing further recursion.
+
+    If traversing into even deeper multipart form hierarchy was desired, we
```
Comment (kgriffs) [inline docs/user/recipes/multipart-mixed.rst:38]: This sentence would better agree with the preceding one if it were converted to present tense, e.g.:

> If traversing into even deeper multipart form hierarchy is desired, we can just reuse the same parser.

File: docs/user/recipes/multipart-mixed.rst [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    could just reuse the same parser.
+
+Let us now use the nested form enabled parser in an app:
```
Comment (kgriffs) [inline docs/user/recipes/multipart-mixed.rst:41]: Using three modifiers for "parser" makes the sentence a little difficult to read. Perhaps reword to "nesting-aware parser" or similar?

File: docs/user/recipes/multipart-mixed.rst [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+    parser = falcon.media.MultipartFormHandler()
+    parser.parse_options.media_handlers['multipart/mixed'] = (
+        falcon.media.MultipartFormHandler())
+
+    app = falcon.App()
+    app.req_options.media_handlers[falcon.MEDIA_MULTIPART] = parser
+    app.add_route('/forms', Forms())
+
+We should now be able to consume a legacy, the now obsolete
```
Comment (kgriffs) [inline docs/user/recipes/multipart-mixed.rst:67]: Seems like this sentence somehow became convoluted.

File: docs/user/recipes/multipart-mixed.rst (hunk 1 above)
Comment (vytas7) [reply docs/user/recipes/multipart-mixed.rst:38]: Indeed, I've just happened to glue those parts without rereading the paragraph.

**Pull request merged**